### PR TITLE
Document the other ech_outer_extensions DoS vector.

### LINKED ISSUE #378: Naive outer_extensions decoding is a DoS risk
A naive implementation of decoding the inner ClientHello can easily go quadratic when processing outer_extensions. Thus far, nothing in TLS has required searching for O(N) extensions, only O(implemented extensions). It's likely a lot of TLS implementations handle the extensions list linearly most of the time.

outer_extensions is now the first time you have to search for extensions you may not have implemented. A naive implementation would then go quadratic, which is a server DoS risk. It doesn't need to go quadratic (for instance, you can sort ClientHelloOuter's extensions and then binary search per lookup), but it's probably worth at least calling this out, if not switching to a simpler decoding scheme.

(For a simpler decoder, we already require CHInner group the compressed extensions consecutively. We could do the same with CHOuter, at which point we only need to reference one byte slice, be it by offset/length or some goofier scheme.)

### PR DESCRIPTION
**Pull request by davidben**
I forgot we still had this open. Closes #378.

### CODE REVIEW

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -435,25 +435,36 @@ The client-facing server computes ClientHelloInner by reversing this process.
...
+  M is the number of ClientHelloOuter extensions ClientHelloOuter and N is the
+  size of OuterExtensions. This problem can be mitigated with a more efficient
+  lookup, such as sorting ClientHelloOuter extensions and using binary search.
+
+* If OuterExtensions contains duplicate extensions, an attacker could cause the
+  server to construct a large ClientHelloInner by including a large
+  extension in ClientHelloOuter, of length L, and an OuterExtensions list
+  referencing N copies of that extension. The server would then use O(N\*L)
+  memory in response to O(N+L) bandwidth from the client. This can be mitigated
+  by checking for duplicates before copying extensions.
```
[the remark this answers is not in the document]
Comment (davidben) [reply draft-ietf-tls-esni.md:467]: As an alternate to this PR, I put together https://github.com/tlswg/draft-ietf-tls-esni/pull/455 which implements this suggestion. Thoughts? At the risk of yet another protocol change in draft-11, I'm starting to lean towards that option.

**Review by cjpatton — APPROVED** [review]

**Comment by davidben** [comment]
From the discussion in https://github.com/tlswg/draft-ietf-tls-esni/pull/455, it sounds like folks seem to agree that version is preferable. (I also prefer it over this.)

**Comment by chris-wood** [comment]
@davidben shall we close this out, then?

**Comment by davidben** [comment]
Sure.

**Pull request closed without merging**